### ISSUE TITLE
Redundant zero factor changes the quadrature family and exposes a 0.004 rank-one pricing error

### ISSUE THREAD

**Issue by microprediction** [microprediction/winning#131]
## Summary

Automatic factor-node selection uses the raw number of loading columns rather than the effective covariance rank. Appending an exactly zero column therefore changes a mathematically identical rank-one model from the rank-one Gauss-Hermite branch to the multi-factor Sobol branch.

That representation mismatch exposes a substantive accuracy defect immediately below the rank-one GH→quantile threshold: the selected 198-node Gauss-Hermite rule is wrong by total variation `0.00401`, while the redundant-column Sobol result agrees with a deterministic two-million-point one-dimensional reference to `6e-10`.

Affected revision: `ead77b3932e86058195f91b9b2185304d09730a2` (`main`).

## Exact reproducer

`softmax_probabilities` is useful here because every conditional probability is analytic; the only approximation is the factor integral.

```python
import numpy as np
from scipy.special import ndtri
from winning.factor.races import softmax_probabilities, _setup

mu = np.array([
    -.35631338, -.25581218, .80898170, .25405868,
    -.28528677,  .31450349, .07612151, .25463292])
v = np.array([
    10.00615591, -1.52603962, -2.69881429, 7.06815397,
    -3.52850322,  8.59944738, -.97098903,  .36919759])
tau = .35

V1 = v[:, None]
V2 = np.column_stack([v, np.zeros_like(v)])
assert np.array_equal(V1 @ V1.T, V2 @ V2.T)

p1 = softmax_probabilities(mu, tau, V=V1)
p2 = softmax_probabilities(mu, tau, V=V2)

D = np.full(len(mu), np.pi**2 / 6 * tau**2)
a = _setup(mu, V1, D, None, None, "gumbel")
b = _setup(mu, V2, D, None, None, "gumbel")
print(len(a[3]), len(b[3]))
print(p1)
print(p2)
print(.5 * np.abs(p1 - p2).sum())

M = 2_000_000
f = ndtri((np.arange(M) + .5) / M)
z = -(mu[None, :] + f[:, None] * v[None, :]) / tau
z -= z.max(axis=1, keepdims=True)
e = np.exp(z)
ref = (e / e.sum(axis=1, keepdims=True)).mean(axis=0)
print(.5 * np.abs(p1 - ref).sum())
print(.5 * np.abs(p2 - ref).sum())
```

Observed:

```text
automatic node counts: 198, 8192

rank-one GH:
[.47961772, .03889049, .00481199, .00636340,
 .44306618, .01192203, .01124528, .00408291]

same covariance plus a zero column:
[.47594264, .04073796, .00484217, .00675053,
 .44283806, .01181457, .01217247, .00490160]

TV between representations       0.0040106523
TV(rank-one GH, reference)       0.0040106522
TV(redundant-column, reference)  5.7e-10
```

The result is stable from 100,000 through 2,000,000 midpoint-normal reference nodes. At `tau=.35`, sharpness is about `24.70`, so `ceil(8*sharp)=198` remains in the high-order GH branch. At `tau=.30`, sharpness crosses the current threshold and the rank-one quantile branch becomes accurate; accuracy therefore improves discontinuously when the noise is made *smaller*.

## Expected versus actual

Expected: adding a zero factor column leaves (V V^	op), the model, the effective rank, and the answer unchanged. The automatic rule should meet its accuracy intent on either representation.

Actual: a zero column changes the integration family and reveals a silent 0.4-percentage-point total-variation error in the nominal rank-one answer. No warning or error estimate is emitted.

## Impact

This affects every auto-node consumer of `factor.races._setup`, including the forward race and mixed-softmax controls. Equivalent factorizations can select different accuracy/cost regimes, and a caller can obtain a more accurate result accidentally by padding `V` with zeros.

## Suggested fix and regression

1. Center and reduce `V` to its numerical effective rank before computing `r`, sharpness, or selecting a node family; exactly zero/dead columns must not count.
2. Move the rank-one near-step crossover below this failing regime, or use a nested-rule/error check rather than assuming high-order GH remains accurate until `Q > 201`.
3. Add a mixed-softmax regression comparing `V` with `column_stack([V, 0])`.
4. Check the rank-one answer against a dense deterministic 1-D normal-quantile reference on both sides of the family threshold.

This is related in theme to #130 (factor-basis invariance), but it is a separate mechanism: raw-column rank changes the automatic quadrature family, and the reference identifies the rank-one GH branch as the inaccurate one.

**Comment by microprediction** [comment]
This also reaches an exported ratings-used path without changing the factor rank. On the same 8-runner rank-1 case, `win_probabilities_factor(..., points=501, return_total=True)` returns raw total `0.9919360`, then silently normalizes it. The normalized probability vector differs from the converged 2001+ point result by TV `0.00619` (largest coordinate error `0.00483`).

`ratings/full.py` and `ratings/nway.py` call this path, and no warning is emitted unless a caller explicitly asks for and inspects the hidden total. The fix/regression should therefore cover the exported factor function and ratings callers: an ~8e-3 mass defect should trigger refinement/failure rather than silent normalization.

**Comment by microprediction** [comment]
PR #152 moves the fixed handover to `Q > 80`, which fixes #131's original field but does not make the crossover reliable. At head `120edf036049aa7c9cc5e6ee55fd50c865b85900`, I found a legal three-runner mixed-logit counterexample where the new rule is a 91x accuracy regression relative to the GH rule it replaces:

```python
mu = np.array([-0.03324208075629076,
                1.6517729928781415,
               -1.6185309121218514])
v = np.array([0.8402325301867833,
              -1.0,
               0.1597674698132174])
tau = 0.08556565743844088
p = softmax_probabilities(mu, tau, V=v[:, None])
```

The centered loading has sharpness `12.886686`, so #152 selects a 104-node midpoint-normal grid. A deterministic `scipy.integrate.quad_vec` integral over `[-12,12]` at `epsabs=epsrel=1e-13` gives

```text
reference   [0.01157922005, 0.00259762418, 0.98582315577]
#152 auto   [0.01172379463, 0.00040725630, 0.98786894907]
old GH-104  [0.01158640254, 0.00261443566, 0.98579916180]
```

Maximum/TV error is `0.00219036788` for the new midpoint rule versus `0.00002399398` for GH-104. The quadrature error estimate is below `1e-13` and the public auto path selects exactly 104 equal weights, so the comparison isolates the node-family decision.

This confirms #131's recommended nested-rule/error check is essential: neither `Q > 201` nor `Q > 80` is a uniformly valid scalar crossover. A single centered random field cannot establish the claimed handover. Please keep #131 open and add tail-crossing mixed-logit cases like this to the regression sweep.

**Comment by microprediction** [comment]
PR #152 has merged as `7eb2cab4b80a0a820ab1417dc8a8fb2b0fd8b99a`; comparison with reviewed head `120edf0360` shows no merge-time file changes. The three-runner mixed-logit counterexample above therefore reproduces on current `main` unchanged: the automatic midpoint-104 result has TV/max error `0.00219037`, versus `2.3994e-05` for GH-104. The fixed-crossover defect is now shipped; #131 remains open.
